# Bonus: Temperature Analysis I

In [92]:
import pandas as pd
from datetime import datetime as dt

In [93]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [94]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [95]:
# Set the date column as the DataFrame index
#df.set_index("date", inplace=True)
#df

### Compare June and December data across all years 

In [102]:
from scipy import stats

In [103]:
# Filter data for June
june = df['date'].dt.month.eq(6)
june_filtered=df[june]
june_filtered

station       date  prcp  tobs
133    USC00519397 2010-06-01  0.00    78
134    USC00519397 2010-06-02  0.01    76
135    USC00519397 2010-06-03  0.00    78
136    USC00519397 2010-06-04  0.00    76
137    USC00519397 2010-06-05  0.00    77
...            ...        ...   ...   ...
19492  USC00516128 2017-06-26  0.02    79
19493  USC00516128 2017-06-27  0.10    74
19494  USC00516128 2017-06-28  0.02    74
19495  USC00516128 2017-06-29  0.04    76
19496  USC00516128 2017-06-30  0.20    75

[1700 rows x 4 columns]

In [59]:
# Filter data for December
dec = df['date'].dt.month.eq(12)
dec_filtered=df[dec]
dec_filtered

station       date  prcp  tobs
305    USC00519397 2010-12-01  0.04    76
306    USC00519397 2010-12-03  0.00    74
307    USC00519397 2010-12-04  0.00    74
308    USC00519397 2010-12-06  0.00    64
309    USC00519397 2010-12-07  0.00    64
...            ...        ...   ...   ...
19323  USC00516128 2016-12-27  0.14    71
19324  USC00516128 2016-12-28  0.14    71
19325  USC00516128 2016-12-29  1.03    69
19326  USC00516128 2016-12-30  2.37    65
19327  USC00516128 2016-12-31  0.90    65

[1517 rows x 4 columns]

In [116]:
# Identify the average temperature for June
june_groupby = june_filtered.groupby(['station'])
june_all = june_groupby['tobs'].mean()
june_all

station
USC00511918    74.139394
USC00513117    74.050847
USC00514830    76.005376
USC00516128    71.937220
USC00517948    76.655405
USC00518838    73.394737
USC00519281    73.271186
USC00519397    77.559322
USC00519523    76.668103
Name: tobs, dtype: float64

In [121]:
june_filtered['tobs'].mean()

74.94411764705882

In [119]:
# Identify the average temperature for December
dec_groupby = dec_filtered.groupby(['station'])
dec_all = dec_groupby['tobs'].mean()
dec_all

station
USC00511918    69.684211
USC00513117    71.069444
USC00514830    73.224719
USC00516128    69.291262
USC00517948    71.834862
USC00518838    72.421053
USC00519281    69.903226
USC00519397    71.109524
USC00519523    72.433333
Name: tobs, dtype: float64

In [122]:
dec_filtered['tobs'].mean()

71.04152933421226

In [123]:
# Create collections of temperature data
june_all_temps = june_filtered.tobs
dec_all_temps = dec_filtered.tobs

In [124]:
# Run paired t-test
stats.ttest_ind(june_all_temps, dec_all_temps)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

The t-test tells us that the two datasets are different, while the pvalue tells us that the results of this test are statistically significant. There is a 3.9 degree difference between June temperatures and December temperatures across all years and weather stations. In conclusion our t-test tells us that this difference is significant. 